api/utils/net/ports.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2024 Gravitational, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package net
+
+import (
+	"github.com/gravitational/trace"
+)
+
+// ValidatePortRange checks if the given port range is within bounds. If endPort is not zero, it
+// also checks if it's bigger than port. A port range with zero as endPort is assumed to describe a
+// single port.
+func ValidatePortRange(port, endPort int) error {
```

Comment:
@marcoandredinis Could you take a look at c931742 which extracts port range validation to `api/utils/net.ValidatePortRange`? I want to be sure that whatever I'm pushing to api looks perfect.